src/libstd/io/mod.rs

```diff
@@ -1461,12 +1460,13 @@ pub trait Write {
     /// ```
     #[unstable(feature = "write_all_vectored", issue = "70436")]
     fn write_all_vectored(&mut self, mut bufs: &mut [IoSlice<'_>]) -> Result<()> {
+        bufs = IoSlice::advance(bufs, 0);
```

Comment:
Let’s add a comment here explaining why this is needed.

Reply:
Added a comment explaining the motivation behind the initial advance.